lib/pbench/cli/server/user_management.py

```diff
@@ -0,0 +1,205 @@
+@cli.command()
+@common_options
+@click.argument("username")
+@pass_cli_context
+def user_delete(context: object, username: str) -> None:
+    context.username = username
+
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-user-delete")
+
+        # Delete the the user with specified username
+        User.delete(username=context.username)
+
+        click.echo(f"User {context.username} deleted")
+        rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# Users list CLI
+@cli.command()
+@common_options
+@pass_cli_context
+def user_list(context: object) -> None:
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-users-list")
+
+        ROW_FORMAT = "{10} {10} {%Y-%m-%d} {}"
+        click.echo(
+            ROW_FORMAT.format("First Name", "Last Name", "Registered On", "Email")
+        )
+
+        # Query all the users
+        users = Database.db_session.query(User).all()
+        for user in users:
+            click.echo(
+                ROW_FORMAT.format(
+                    user.first_name, user.last_name, user.registered_on, user.email
+                )
+            )
+
+        rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# User update CLI
+@cli.command()
+@common_options
+@click.argument("user-to-update")
+@click.option(
+    "--username", required=False, help="Specify the new username",
+)
+@click.option(
+    "--email", required=False, help="Specify the new email",
+)
+@click.option(
+    "--first-name", required=False, help="Specify the new first name",
+)
+@click.option(
+    "--last-name", required=False, help="Specify the new last name",
+)
+@click.option(
+    "--role", required=False, help="Specify the new role",
+)
+@pass_cli_context
+def user_update(
+    context: object,
+    user_to_update: str,
+    username: str,
+    first_name: str,
+    last_name: str,
+    email: str,
+    role: str,
+) -> None:
+    try:
+        # Setup the pbench server config and db access
+        config_setup(context, "pbench-user-update")
+
+        # Query the user
+        user = User.query(username=user_to_update)
+
+        if user is None:
+            click.echo(f"No such a user {user_to_update} to update")
+            rv = 1
+        else:
+            dict_to_update = {}
+            if username:
+                dict_to_update["username"] = username
+
+            if first_name:
+                dict_to_update["first_name"] = first_name
+
+            if last_name:
+                dict_to_update["last_name"] = last_name
+
+            if email:
+                dict_to_update["email"] = email
+
+            if role:
+                dict_to_update["role"] = role
```

Comment:
If `role` is omitted, does that leave it unchanged?  If so, how do we remove a role?  (On the flip side, does specifying one replace existing role(s) or add it?)

(This is probably not a high-priority problem, since I don't expect that we're going to want to modify the Admin user, but it's something to ponder going forward....)

Reply:
Specifying one would replace the existing one rn, since we are not registering it as list in our table it's just one value. However, that is an interesting point to consider, we may want role as a list instead of just one value. That however we can take care when we fully try to implement role based authentication model (whenever we get to that point).

Reply:
Yes -- I don't think we need to take any action on this for this PR -- however, as presented, there seems to be no way to remove a role once it is set.  I don't expect this to be a problem initially, since this tool will only be used internally to create the default users (regular and admin), but we will probably want a little more capability here, just in case someone makes a mistake with this tool....